datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -190,6 +190,12 @@ impl TableProviderFactory for ListingTableFactory {
             .with_definition(cmd.definition.clone())
             .with_constraints(cmd.constraints.clone())
             .with_column_defaults(cmd.column_defaults.clone());
+
+        // Pre-warm statistics cache if collect_statistics is enabled
+        if session_state.config().collect_statistics() {
+            let _ = table.list_files_for_scan(state, &[], None).await?;
```

Comment:
Should failures in the pre-warming be propagated ?
Maybe handle/ignore failures locally ?!